hw2/task01/task01.py

```diff
@@ -0,0 +1,87 @@
+"""
+Given a file containing text.
+
+Complete using only default collections:
+    1) Find 10 longest words consisting from largest amount of unique symbols
+    2) Find rarest symbol for document
+    3) Count every punctuation char
+    4) Count every non ascii char
+    5) Find most common non ascii char for document
+"""
+
+import string
+from unicodedata import category
+
+
+def open_file_and_count_chars(file_path: str) -> dict:
+    """Open file and decipher and count chars."""
+    counter = {}
+    with open(file_path, "r", encoding="unicode-escape") as text:
+        for line in text:
+            for char in line:
+                if char in counter:
+                    counter[char] += 1
+                else:
+                    counter[char] = 1
+    return counter
+
+
+def open_file_and_yield_line(file_path: str) -> str:
+    with open(file_path, "r", encoding="unicode-escape") as text:
+        for line in text:
+            yield line
```

Comment:
What is wrong with this function? (it is a simple thing that can't be watched by tests)